app/src/main/kotlin/compiler/syntax/AstFactory.kt

```diff
@@ -409,6 +411,60 @@ class AstFactory(private val diagnostics: Diagnostics) {
                 val bodyBlock = processMaybeBlock(children[4])
                 Statement.Loop(conditionExpr, bodyBlock, combineLocations(children))
             }
+            in listOf(Productions.nonBraceStatementArrayLoop, Productions.nonIfNonBraceStatementArrayLoop) -> {
+                val arrayExpression = processExpression(children[6])
```

Comment:
Please extract this code into a method (e.g. `makeArrayLoop`) which could also be used in #179.